### PR TITLE
JsonCodec Act 2

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Now DataObject doesn't generate a codec

Related to #260

Signed-off-by: slinkydeveloper <francescoguard@gmail.com>

### CODE REVIEW

File: src/main/java/io/vertx/codegen/type/DataObjectTypeInfo.java [hunk 1]
```diff
@@ -3,85 +3,151 @@
...
   }
 
-  public String getJsonEncoderFQCN() {
-    if (hasJsonEncoder())
-      if (getJsonEncoderEnclosingClass() != null)
-        return jsonEncoderPackage + "." + getJsonEncoderEnclosingClass() + "." + jsonEncoderSimpleName;
-      else
-        return jsonEncoderPackage + "." + jsonEncoderSimpleName;
-    else return null;
+  public static class DataObjectAnnotatedInfo {
```
Comment (vietj) [inline src/main/java/io/vertx/codegen/type/DataObjectTypeInfo.java]: can you make this type and the other not inner class ? we might reuse  "DataObjectAnnotatedInfo" these in the future to have `@VertxGen` also being data object

**Comment by okou19900722** [comment]
does `JsonCodec` of `DataObjectConverter` useful? I saw that it's never use again

**Comment by slinkydeveloper** [comment]
@okou19900722 It will be used later for runtime encoding/decoding of json, look here: https://github.com/vert-x3/wiki/wiki/RFC:JSON-improvements#iovertxcorejson

**Comment by okou19900722** [comment]
@slinkydeveloper yes, i had saw. an other issues, how can i judge the encode method of DataObject annotated with @DataObject is toJson or Converter.toJson

**Comment by okou19900722** [comment]
I saw that it's always toJson of the model but not toJson of the converter
https://github.com/vert-x3/vertx-lang-js/blob/master/vertx-lang-js-gen/src/main/resources/vertx-js/util/utils.js#L476-L480

**Comment by slinkydeveloper** [comment]
Yes codegen still uses `toJson()` and constructor with `JsonObject`

JsonCodec will be used both at compile time by the codegen and at runtime by new classes i'm working on that will land on core

**Comment by okou19900722** [comment]
I means how to decide the encode method of a type which annotated with @DataObject. no toJson in the type.

**Comment by okou19900722** [comment]
https://github.com/vert-x3/vertx-codegen/blob/9f6d328437c18f60c43e58357d8efa0449139540/src/main/java/io/vertx/codegen/Helper.java#L682

`isDataObjectAnnotatedEncodable` depends on whether a Converter is generated or has a toJson method

**Comment by okou19900722** [comment]
```js
utils.convReturnDataObjectAnnotated = function(dataObject) {
  if (dataObject != null) {
    return utils.convReturnJson(dataObject.toJson());
  }
  return null;
};
```
no `toJson` in dataObject. maybe I need to invoke `XXXConverter.toJson(dataObject, json)`.

**Comment by slinkydeveloper** [comment]
yes yes we just noticed this bug and i opened a pr for this :smile: https://github.com/vert-x3/vertx-codegen/pull/263

**Pull request merged**